java/statestore/src/main/java/sleeper/statestore/s3/S3FileReferenceStore.java

```diff
@@ -120,6 +121,59 @@ public void addFiles(List<FileReference> fileReferences) throws StateStoreExcept
         }
     }
 
+    @Override
+    public void splitFileReferences(List<SplitFileReferenceRequest> splitRequests) throws StateStoreException {
+        Instant updateTime = clock.instant();
+        for (SplitFileReferenceRequest splitRequest : splitRequests) {
+            splitFileReference(splitRequest, updateTime);
```

Comment:
One reason for the API accepting a list of requests is that the state store can optimise the execution of those by doing them in a single batch. The implementation here makes multiple updates to the state store, each requiring reading the current state from S3, updating it and then attempting to write it back to S3 before any other process can update it. It would be more efficient to make all the changes in one go.

Reply:
Thanks, these are now all done in 1 S3 update